hydra-cluster/src/Hydra/Generator.hs

```diff
@@ -87,34 +94,22 @@ generateConstantUTxODataset ::
   IO Dataset
 generateConstantUTxODataset nClients nTxs = do
   (_, faucetSk) <- keysFor Faucet
-  generate $ genDatasetConstantUTxO faucetSk nClients nTxs
+  clientKeys <- generate $ replicateM nClients arbitrary
+  fundingTransaction <- generate $ makeGenesisFundingTx faucetSk clientKeys
+  generate $ genDatasetConstantUTxO clientKeys nTxs fundingTransaction
 
 genDatasetConstantUTxO ::
-  -- | The faucet signing key
-  SigningKey PaymentKey ->
-  -- | Number of clients
-  Int ->
+  -- | Clients
+  [ClientKeys] ->
```

Comment:
Should: not use `ClientKeys` here. This type is strictly bigger than what we need. A list of cardano signing keys should be enough here. i.e. `[SigningKey PaymentKey]`

Reply:
The problem is that the function has to Generate a Dataset and for that it needs [ClientKeys]

Reply:
The comment was about that `ClientKeys` contains two signing keys. But we only need one of them (iirc it was called `externalSigningKey`) to fund the client.